### PR TITLE
samples: matter: Adjust Matter samples to PSA SSF Crypto Service

### PR DESCRIPTION
**Pull request by ArekBalysNordic**
Adjusted kconfigs selected by OPENTHREAD_NRF_SECURITY_PSA to work with PSA SSF Crypto Client on nRF54H20.
Updated DTS files to disable fake PRNG and enable entropy from SSF crypto service.
Enabled SSF Crypto Service in config overlays.

### CODE REVIEW

File: subsys/net/openthread/Kconfig [hunk 1]
```diff
@@ -64,7 +65,7 @@ config OPENTHREAD_NRF_SECURITY_PSA
 
 config OPENTHREAD_NRF_SECURITY_PSA
 	bool
-	select MBEDTLS_PSA_CRYPTO_STORAGE_C
+	select MBEDTLS_PSA_CRYPTO_STORAGE_C if !PSA_SSF_CRYPTO_CLIENT
 	select TRUSTED_STORAGE
```
Comment (LuDuda) [inline subsys/net/openthread/Kconfig:69]: not related much.. but should we enable Trusted Storage for TF-M or `PSA_SSF_CRYPTO_CLIENT` (nRF54H)? Not sure if we use `OPENTHREAD_NRF_SECURITY_PSA` anywhere in Matter.

**Review by LuDuda — APPROVED** [review]

**Review by LuDuda — APPROVED** [review]

**Review by maciejbaczmanski — APPROVED** [review]

File: samples/matter/common/dts/nrf54h20/nrf54h20_cpuapp_peripherals.dtsi [hunk 2]
```diff
@@ -24,7 +24,7 @@
 	status = "okay";
 };
 
-/* Communication between Application Core and Secure domain for SUIT */
+/* Communication between Application Core and Secure domain */
```
Comment (greg-fer) [inline samples/matter/common/dts/nrf54h20/nrf54h20_cpuapp_peripherals.dtsi:27]:
```suggestion
/* Communication between Application Core and Secure Domain */
```

File: samples/matter/common/src/persistent_storage/Kconfig [hunk 3]
```diff
@@ -13,13 +13,14 @@ config NCS_SAMPLE_MATTER_SETTINGS_STORAGE_BACKEND
...
 config NCS_SAMPLE_MATTER_SECURE_STORAGE_BACKEND
 	bool "Secure storage implementation for Matter samples"
-	select TRUSTED_STORAGE if !BUILD_WITH_TFM
-	select PSA_PROTECTED_STORAGE if !BUILD_WITH_TFM
+	select TRUSTED_STORAGE if (!PSA_SSF_CRYPTO_CLIENT && !BUILD_WITH_TFM)
+	select PSA_PROTECTED_STORAGE if (!PSA_SSF_CRYPTO_CLIENT && !BUILD_WITH_TFM)
 	help
 	  Enables secure persistent storage wrapper API, that
 	  imitates Zephyr Settings key-value data format.
 	  If building with CMSE enabled (*/ns), the TF-M
```
Comment (greg-fer) [inline samples/matter/common/src/persistent_storage/Kconfig:21]:
```suggestion
	  Enables the secure persistent storage wrapper API that
	  imitates Zephyr Settings' key-value data format.
	  If building with CMSE enabled (*/ns), the TF-M
```

File: samples/matter/common/src/persistent_storage/Kconfig [hunk 4]
```diff
@@ -13,13 +13,14 @@ config NCS_SAMPLE_MATTER_SETTINGS_STORAGE_BACKEND
...
-	  PS implementation is leveraged by default.
+	  and Secure Domain PS implementation is leveraged
```
Comment (greg-fer) [inline samples/matter/common/src/persistent_storage/Kconfig:22]: Please expand PS here.

File: samples/matter/common/src/persistent_storage/Kconfig [hunk 5]
```diff
@@ -13,13 +13,14 @@ config NCS_SAMPLE_MATTER_SETTINGS_STORAGE_BACKEND
...
+	  by default.
 	  If building with CMSE disabled (cpuapp target),
```
Comment (greg-fer) [inline samples/matter/common/src/persistent_storage/Kconfig:24]: Minor alignment?
```suggestion
	  If building with CMSE disabled (*/cpuapp),
```

File: samples/openthread/cli/boards/nrf54h20dk_nrf54h20_cpuapp.overlay [hunk 6]
```diff
@@ -43,3 +43,11 @@
 		};
 	};
 };
+
+/* Communication between Application Core and Secure domain */
```
Comment (greg-fer) [inline samples/openthread/cli/boards/nrf54h20dk_nrf54h20_cpuapp.overlay:47]:
```suggestion
/* Communication between Application Core and Secure Domain */
```

**Review by greg-fer — APPROVED** [review]

**Pull request merged**